From roel.kluin@gmail.com  Tue Aug 25 22:15:48 2009
From: Roel Kluin <roel.kluin@gmail.com>
Date: Tue, 25 Aug 2009 10:26:57 +0200
Subject: USB: double put_tty_driver(gs_tty_driver) in gserial_setup()
To: David Brownell <dbrownell@users.sourceforge.net>, linux-usb@vger.kernel.org, Andrew Morton <akpm@linux-foundation.org>
Message-ID: <4A93A051.5060506@gmail.com>


If the driver cannot be registered, put_tty_driver(gs_tty_driver)
occurred here as well as at label fail.

put_tty_driver() already occurs at label fail

Signed-off-by: Roel Kluin <roel.kluin@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/u_serial.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/usb/gadget/u_serial.c
+++ b/drivers/usb/gadget/u_serial.c
@@ -1114,7 +1114,6 @@ int __init gserial_setup(struct usb_gadg
 	/* export the driver ... */
 	status = tty_register_driver(gs_tty_driver);
 	if (status) {
-		put_tty_driver(gs_tty_driver);
 		pr_err("%s: cannot register, err %d\n",
 				__func__, status);
 		goto fail;
